lib/internal/perf/observe.js

```diff
@@ -372,6 +384,32 @@ function enqueue(entry) {
   }
 }
 
+/**
+ * Add the resource timing entry to the global buffer if the buffer size is not
+ * exceeding the buffer limit, or dispatch a buffer full event on the global
+ * performance object.
+ */
+function bufferResourceTiming(entry) {
+  if (resourceTimingBuffer.length >= resourceTimingBufferSizeLimit) {
+    dispatchBufferFull('resourcetimingbufferfull');
```

Comment:
Not too familiar with the spec, but is this timing expected? Seems to me like it makes more sense for the event to dispatch immediately after reaching capacity so the buffer can be consumed before the next event is received, not at the point the next event is received and discarding the event without having any chance to react before it is discarded. 🤔

Reply:
Yeah, seems like this is a mismatch with the spec. I'll check again on the steps with the spec conformance. Thanks for pointing this out!

Reply:
Updated with aligning the behavior with https://www.w3.org/TR/resource-timing-2/#dfn-add-a-performanceresourcetiming-entry.